dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -0,0 +1,543 @@
+import com.dotcms.workflow.helper.WorkflowHelper;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowStep;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.Beta;
+import com.liferay.portal.model.User;
+import com.liferay.util.LocaleUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Locale;
+
+@SuppressWarnings("serial")
+@Beta /* Non Official released */
+@Path("/v1/workflow")
+public class WorkflowResource {
+
+    private final WorkflowHelper workflowHelper;
+    private final WebResource webResource;
+    private final WorkflowAPI workflowAPI;
+    private final ResponseUtil responseUtil;
+
+
+    /**
+     * Default constructor.
+     */
+    public WorkflowResource() {
+        this(WorkflowHelper.getInstance(),
+                APILocator.getWorkflowAPI(),
+                ResponseUtil.INSTANCE,
+                new WebResource());
+    }
+
+    @VisibleForTesting
+    protected WorkflowResource(final WorkflowHelper workflowHelper,
+                               final WorkflowAPI workflowAPI,
+                               final ResponseUtil responseUtil,
+                               final WebResource webResource) {
+
+        this.workflowHelper = workflowHelper;
+        this.webResource    = webResource;
+        this.responseUtil   = responseUtil;
+        this.workflowAPI    = workflowAPI;
+    }
+
+
+    /**
+     * Returns a single action associated to the step, 404 if does not exists. 401 if the user does not have permission.
+     * @param request  HttpServletRequest
+     * @param schemeId String
+     * @return Response
+     */
+    @GET
+    @Path("/schemes/{schemeId}/steps")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findStepsByScheme(@Context final HttpServletRequest request,
+                                     @PathParam("schemeId") final String schemeId) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
+        List<WorkflowStep> steps;
+
+        try {
+
+            Logger.debug(this, "Getting the workflow steps for the scheme: " + schemeId);
+            steps     = this.workflowHelper.findSteps(schemeId);
+
+            response  =
+                    Response.ok(new ResponseEntityView(steps)).build(); // 200
+        } catch (DoesNotExistException e) {
+
+            Logger.error(this.getClass(),
+                    "DoesNotExistException on findStepsByScheme, schemeId: " + schemeId +
+                            ", exception message: " + e.getMessage(), e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal ", exception message: " 19 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)